lib/std/os/linux.zig

```diff
@@ -1386,6 +1386,58 @@ pub fn madvise(address: [*]u8, len: usize, advice: u32) usize {
     return syscall3(.madvise, @ptrToInt(address), len, advice);
 }
 
+pub fn posix_fadvise(fd: fd_t, offset: off_t, len: off_t, advice: usize) usize {
+    if (usize_bits == 64 and @hasField(SYS, "fadvise64")) {
+        return syscall4(
+            .fadvise64,
+            @bitCast(usize, @as(isize, fd)),
+            @bitCast(usize, offset),
+            @bitCast(usize, len),
+            advice,
+        );
+    }
+
+    const offset_halves = splitValue64(@bitCast(u64, offset));
+    const length_halves = splitValue64(@bitCast(u64, len));
+
+    if (comptime std.Target.current.cpu.arch.isMIPS()) {
+        // MIPS requires a 7 argument syscall
+
+        return syscall7(
+            .fadvise64,
+            @bitCast(usize, @as(isize, fd)),
+            0,
+            offset_halves[0],
+            offset_halves[1],
+            length_halves[0],
+            length_halves[1],
+            advice,
+        );
+    } else if (comptime std.Target.current.cpu.arch.isARM()) {
+        // ARM reorders the arguments
+
+        return syscall6(
+            .arm_fadvise64_64,
```

Comment:
This is not needed as `arm_fadvise64_64` is aliased to `fadvise64_64`.

Reply:
Addressed in the latest commit